api/v1/controllers/subjects.js

```diff
@@ -249,32 +252,67 @@ module.exports = {
       }
     }
 
-    const findByKeyPromise =
-      featureToggles.isFeatureEnabled(sampleStoreFeature) ?
-            u.findByKey(helper, params, ['subjectHierarchy']) :
-              u.findByKey(helper, params, ['hierarchy', 'samples']);
-    findByKeyPromise
-    .then((o) => {
-      resultObj.dbTime = new Date() - resultObj.reqStartTime;
-      let retval = u.responsify(o, helper, req.method);
-      if (depth > ZERO) {
-        retval = helper.deleteChildren(retval, depth);
-      }
-
-      // if samples are stored in redis, get the samples from there.
-      if (featureToggles.isFeatureEnabled(sampleStoreFeature)) {
-        redisSubjectModel.completeSubjectHierarchy(retval, params)
-        .then((_retval) => {
-          u.logAPI(req, resultObj, _retval);
-          res.status(httpStatus.OK).json(_retval);
-        });
-      } else {
-        retval = helper.modifyAPIResponse(retval, params);
-        u.logAPI(req, resultObj, retval);
-        res.status(httpStatus.OK).json(retval);
-      }
-    })
-    .catch((err) => u.handleError(next, err, helper.modelName));
+    const resultObj = {
+      reqStartTime: Date.now(),
+      params: params,
+    };
+
+    if (featureToggles.isFeatureEnabled('enableWorkerProcess')
+    && featureToggles.isFeatureEnabled('enqueueHierarchy')) {
+      jobWrapper.createJob(jobType.GET_HIERARCHY, resultObj, req)
+      .ttl(WORKER_TTL)
+      .on('complete', (resultObj) => {
+        u.logAPI(req, resultObj, resultObj.retval);
```

Comment:
They are both called. If the listener in jobWrapper isn't called, the logging test will fail.
I think it's better to do this here because the listener in jobWrapper is generic and this one is specific to getHierarchy jobs.